sed/loader/flash/loader.py

```diff
@@ -0,0 +1,589 @@
+"""
+This module implements the flash data loader.
+The raw hdf5 data is saved into parquet files and loaded as a dask dataframe.
+If there are multiple files, the NaNs are forward filled.
+"""
+import os
+from functools import reduce
+from itertools import compress
+from pathlib import Path
+from typing import cast
+from typing import List
+from typing import Sequence
+from typing import Tuple
+from typing import Union
+
+import dask.dataframe as dd
+import h5py
+import numpy as np
+from joblib import delayed
+from joblib import Parallel
+from pandas import DataFrame
+from pandas import MultiIndex
+from pandas import Series
+
+from sed.loader.base.loader import BaseLoader
+from sed.loader.exception_handlers import NoFilesFoundError
+from sed.loader.utils import gather_flash_files
+from sed.loader.utils import parse_h5_keys
+
+
+class FlashLoader(BaseLoader):
+    """
+    The class generates multiindexed multidimensional pandas dataframes
+    from the new FLASH dataformat resolved by both macro and microbunches
+    alongside electrons.
+    """
+
+    __name__ = "flash"
+
+    supported_file_types = ["h5", "parquet"]
+
+    def __init__(self, config: dict, meta_handler) -> None:
+
+        super().__init__(config=config, meta_handler=meta_handler)
+        self._config: dict = config.get("dataframe", {})
+        self.config_parser()
+        # Set all channels, exluding pulseId as default
+        self.all_channels: dict = self._config.get("channels", {})
+        self.index_per_electron: Union[MultiIndex, None] = None
+        self.index_per_pulse: Union[MultiIndex, None] = None
+        self.parquet_names: List[Path] = []
+        self.failed_files_error: List[str] = []
+
+    def config_parser(self):
+        """
+        Parser for the config.yaml file.
+        """
+        paths = self._config.get("paths")
+
+        # Prases to locate the raw beamtime directory from config file
+        if paths:
+            if "data_raw_dir" in paths:
+                self.data_raw_dir = Path(paths["data_raw_dir"])
+            if "data_parquet_dir" in paths:
+                self.data_parquet_dir = Path(paths["data_parquet_dir"])
+                if not self.data_parquet_dir.exists():
+                    os.mkdir(self.data_parquet_dir)
+
+        if not {"ubid_offset", "daq"}.issubset(self._config.keys()):
+            raise ValueError(
+                "One of the values from ubid_offset or daq is missing. \
+                        These are necessary.",
+            )
```

Comment:
You can make this required, however, I would also suggest to allow for the flash loader to also just work with passing it file names or a folder to work on (see below)